src/dscom/TypeInfoResolver.cs

```diff
@@ -67,8 +67,54 @@ public TypeInfoResolver(WriterContext writerContext)
         return typeInfo;
     }
 
+    /// <summary>
+    /// Resolve the <see cref="ITypeInfo"/> by a <see cref="Guid"/>. If the <see cref="ITypeInfo"/> is not
+    /// present yet, it will try to add the type library for the corresponding assembly.
+    /// </summary>
+    private ITypeInfo? ResolveTypeInfo(Type type, Guid guid)
+    {
+        // If the given type is not present in _types
+        // this means, that the typelib is not loaded yet
+        if (_types.TryGetValue(guid, out var typeInfo))
+        {
+            return typeInfo;
+        }
+
+        var assembly = type.Assembly;
+
+        // check if the type library is already present
+        var identifier = GetTypeLibFromIdentifier(assembly.GetLibIdentifier(WriterContext.Options.OverrideTlbId));
+        if (identifier is null)
+        {
+            foreach (var additionalLib in _additionalLibs)
+            {
+                var name = assembly.GetName().Name ?? string.Empty;
```

Comment:
The changes from the following PR should be adopted here:

#328 
#329 

The problem here is, that assumption that the name of assembly is eqal to the name of tlb is incorrect.
The behavior should be, that we first try to get the tlb matching the assembly name.
If there is no tlb with this name, try find it in all other tlbs (see changes in pull request #329 and #328)
This all is necessary because of #310 

...it's complicated